### PR TITLE
Added the java4.x client cluster filter pull test content

### PR DESCRIPTION
**Pull request by Zintoki**
## Brief changelog

Added the java4.x client cluster filter pull test content

## Verifying this change

Content:

 - [x] Test some Settings of 4.x client message, pushconsumer, pullconsumer and producer
 - [x] Test Clust, single producer, multiple pushconsumer reception
 - [x] Test the Tag and the Sql, and find the relevant information content through the tag
 - [x] Test the pullconsumer by using updating offset(message ACK and message NACK)

### CODE REVIEW

File: java/e2e-v4/src/test/java/org/apache/rocketmq/client/producer/ProducerInitTest.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            DefaultMQProducer producer = new DefaultMQProducer(RandomUtils.getStringByUUID(),(RPCHook) null);
+            producer.setNamesrvAddr(namesrvAddr);
+            producer.setInstanceName(UUID.randomUUID().toString());
+            producer.start();
+            producer.shutdown();
+        }, "Expected ClientException to throw, but it didn't");
+    }
+
+    @Disabled
+    @DisplayName("The Producer sets the maximum retry times to 0, expect the client start success")
```
Comment (cryptoya) [inline java/e2e-v4/src/test/java/org/apache/rocketmq/client/producer/ProducerInitTest.java:121]: 用例期望与实现不一致

File: java/e2e-v4/src/test/java/org/apache/rocketmq/cluster/ClusterTest.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        if (pullConsumer != null) {
+            pullConsumer.shutdown();
+        }
+        if (producer != null) {
+            producer.shutdown();
+        }
+    }
+
+    @Test
+    @DisplayName("Send 100 normal messages synchronously, start three consumers on different GroupId, and expect each client to consume up to 100 messages")
```
Comment (cryptoya) [inline java/e2e-v4/src/test/java/org/apache/rocketmq/cluster/ClusterTest.java:72]: 增加一下MessageModel为广播模式的场景

**Review by cryptoya — APPROVED** [review]

**Comment by Zintoki** [comment]
OK



---Original---
From: ***@***.***&gt;
Date: Wed, Oct 18, 2023 16:41 PM
To: ***@***.***&gt;;
Cc: "XiaoWei ***@***.******@***.***&gt;;
Subject: Re: [apache/rocketmq-e2e] Added the java4.x client cluster filterpull test content (PR #51)




 
@cryptoya approved this pull request.
 
 
In java/e2e-v4/src/test/java/org/apache/rocketmq/server/abnormal/PushConsumerRetryTest.java:
 &gt; +                } + +            }); +            pushConsumer.start(); +        } catch (MQClientException e) { +            Assertions.fail(e.getMessage()); +        } + +        Assertions.assertNotNull(producer, "Get Producer Failed"); +        for (int i = 0; i < SEND_NUM; i++) { +            Message message = MessageFactory.buildNormalMessage(topic, tag, String.valueOf(i)); +            producer.send(message); +        } +        Assertions.assertEquals(SEND_NUM, producer.getEnqueueMessages().getDataSize(), "send message failed"); + +        TestUtils.waitForMinutes(2);  
建议缩短等待时间，改成测试最大重试次数为0

**Comment by Zintoki** [comment]
ok



---Original---
From: ***@***.***&gt;
Date: Tue, Oct 17, 2023 15:40 PM
To: ***@***.***&gt;;
Cc: "XiaoWei ***@***.******@***.***&gt;;
Subject: Re: [apache/rocketmq-e2e] Added the java4.x client cluster filterpull test content (PR #51)




 
@cryptoya commented on this pull request.
 
 
In java/e2e-v4/src/test/java/org/apache/rocketmq/cluster/ClusterTest.java:
 &gt; +        if (pushConsumer02 != null) { +            pushConsumer02.shutdown(); +        } +        if (pushConsumer03 != null) { +            pushConsumer03.shutdown(); +        } +        if (pullConsumer != null) { +            pullConsumer.shutdown(); +        } +        if (producer != null) { +            producer.shutdown(); +        } +    } + +    @Test +    @DisplayName("Send 100 normal messages synchronously, start three consumers on different GroupId, and expect each client to consume up to 100 messages")  
增加一下MessageModel为广播模式的场景

**Review by lollipopjin — APPROVED** [review]
LGTM

**Comment by cryptoya** [comment]
LGTM

**Pull request merged**